src/test/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchProcessResponseFilesImplTest.java

```diff
@@ -0,0 +1,256 @@
+
+        // when
+        detsToArmBatchProcessResponseFilesImpl.onUploadFileChecksumValidationSuccess(batchInputUploadFileFilenameProcessor,
+                                                                                     batchData,
+                                                                                     uploadFileRecord,
+                                                                                     externalObjectDirectoryEntity,
+                                                                                     checksum);
+
+        // then
+        verify(osrRepository, times(1)).findByArmEodId(anyString());
+        verify(osrRepository, times(1)).save(objectStateRecordEntity);
+    }
+
+    @Test
+    void onUploadFileChecksumValidationFailure() {
+        // given
+        ArmResponseUploadFileRecord uploadFileRecord = new ArmResponseUploadFileRecord();
+        uploadFileRecord.setA360FileId("a360FileId");
+        uploadFileRecord.setMd5("invalidchecksum");
+        when(osrRepository.findByArmEodId(anyString())).thenReturn(Optional.of(objectStateRecordEntity));
+
+        // when
+        detsToArmBatchProcessResponseFilesImpl
+            .onUploadFileChecksumValidationFailure(uploadFileRecord, externalObjectDirectoryEntity, "checksum");
+
+        // then
+        verify(osrRepository, times(1)).findByArmEodId(anyString());
+        verify(osrRepository, times(1)).save(objectStateRecordEntity);
+    }
+
+    @Test
+    void processUploadFileDataFailure() {
+        // given
+        ArmResponseUploadFileRecord uploadFileRecord = mock(ArmResponseUploadFileRecord.class);
+        UploadFileFilenameProcessor processor = mock(UploadFileFilenameProcessor.class);
+        when(osrRepository.findByArmEodId(anyString())).thenReturn(Optional.of(objectStateRecordEntity));
+
+        // when
+        detsToArmBatchProcessResponseFilesImpl.processUploadFileDataFailure(uploadFileRecord, processor, externalObjectDirectoryEntity);
+
+        // then
+        verify(osrRepository, times(1)).findByArmEodId(anyString());
+        verify(osrRepository, times(1)).save(objectStateRecordEntity);
+    }
+
+    @Test
+    void processInvalidLineFileActions() {
+        // given
+        ArmResponseInvalidLineRecord invalidLineRecord = new ArmResponseInvalidLineRecord();
+        invalidLineRecord.setErrorStatus("errorStatus");
+        invalidLineRecord.setExceptionDescription("exceptionDescription");
+        invalidLineRecord.setInput(
+            "{\\\"operation\\\":\\\"create_record\\\",\\\"relation_id\\\":\\\"<EODID>\\\","
+                + "\\\"record_metadata\\\":{\\\"record_class\\\":\\\"A360TEST\\\",\\\"publisher\\\":\\\"A360\\\","
+                + "\\\"region\\\":\\\"GBR\\\",\\\"title\\\":\\\"CGITestFilesMalformedManifest_1.00MB_100.00MB_001\\\","
+                + "\\\"recordDate\\\":\\\"2023-12-21T10:03:53Z\\\"}}");
```

Comment:
Optional: use """ instead of " to allow for easier reading of the text (would prevent the need for double escapes, etc

Reply:
done